## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#94890947-bfba-4565-9984-e74e6bff7b57) |
<a name="94890947-bfba-4565-9984-e74e6bff7b57"></a>
### 11024-32c2b05/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31994
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b01eb627c873:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-tlpd6pxsoip361",
    "objectName1": "object1.csv",
    "objectName2": "object2.csv"
  },
  "duration": 922,
  "error": "ConditionNotMet: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:5589354f-601e-000e-7d65-d201e5000000\nTime:2020-12-14T22:08:00.6370731Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/aws-sdk-go-test-tlpd6pxsoip361/object1.csv?timeout=1501\n   Authorization: REDACTED\n   If-Match: [41455327d3ce43e6c1d1fec0473922e3]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [1ac63ef2-0f2d-4563-5733-bd0e106cb518]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:08:00 GMT]\n   X-Ms-Range: [bytes=0-553]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:08:00 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [1ac63ef2-0f2d-4563-5733-bd0e106cb518]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [5589354f-601e-000e-7d65-d201e5000000]\n   X-Ms-Version: [2019-02-02]\n\n\n\n\tstatus code: 412, request id: 1650B4A3424518AB, host id: ",
  "function": "testSelectObject",
  "message": "AWS SDK Go Select object failed ConditionNotMet: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:5589354f-601e-000e-7d65-d201e5000000\nTime:2020-12-14T22:08:00.6370731Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/aws-sdk-go-test-tlpd6pxsoip361/object1.csv?timeout=1501\n   Authorization: REDACTED\n   If-Match: [41455327d3ce43e6c1d1fec0473922e3]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [1ac63ef2-0f2d-4563-5733-bd0e106cb518]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:08:00 GMT]\n   X-Ms-Range: [bytes=0-553]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:08:00 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [1ac63ef2-0f2d-4563-5733-bd0e106cb518]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [5589354f-601e-000e-7d65-d201e5000000]\n   X-Ms-Version: [2019-02-02]\n\n\n\n\tstatus code: 412, request id: 1650B4A3424518AB, host id: ",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-php",
  "function": "listMultipartUploads ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5fd7e24286b7e",
    "Object": "obj1"
  },
  "duration": "757",
  "status": "FAIL",
  "error": "Error executing \"UploadPartCopy\" on \"http://minio-dev7.minio.io:31994/aws-sdk-php-5fd7e24286b7e/obj1-copy?uploadId=53273dda077fc9fb&partNumber=1\"; AWS HTTP error: Client error: `PUT http://minio-dev7.minio.io:31994/aws-sdk-php-5fd7e24286b7e/obj1-copy?uploadId=53273dda077fc9fb&partNumber=1` resulted in a `412 Precondition Failed` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>ConditionNotMet</Code><Message>-&gt; github.com/Azure/azure-storage- (truncated...)\n ConditionNotMet (client): -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:1bd3a78a-c01e-0148-5c65-d29926000000\nTime:2020-12-14T22:08:08.9917572Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/aws-sdk-php-5fd7e24286b7e/obj1?timeout=1501\n   Authorization: REDACTED\n   If-Match: [c21b73aa229ee10d81eee0aa2044d4a8]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [0dc2b9bc-e974-494d-478c-6da6d0d1c54f]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:08:08 GMT]\n   X-Ms-Range: [bytes=0-1023]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:08:08 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [0dc2b9bc-e974-494d-478c-6da6d0d1c54f]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [1bd3a78a-c01e-0148-5c65-d29926000000]\n   X-Ms-Version: [2019-02-02]\n\n\n - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>ConditionNotMet</Code><Message>-&gt; github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42&#xA;===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====&#xA;Description=The condition specified using HTTP conditional header(s) is not met.&#xA;RequestId:1bd3a78a-c01e-0148-5c65-d29926000000&#xA;Time:2020-12-14T22:08:08.9917572Z, Details: &#xA;   Code: ConditionNotMet&#xA;   GET https://minioazure.blob.core.windows.net/aws-sdk-php-5fd7e24286b7e/obj1?timeout=1501&#xA;   Authorization: REDACTED&#xA;   If-Match: [c21b73aa229ee10d81eee0aa2044d4a8]&#xA;   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]&#xA;   X-Ms-Client-Request-Id: [0dc2b9bc-e974-494d-478c-6da6d0d1c54f]&#xA;   X-Ms-Date: [Mon, 14 Dec 2020 22:08:08 GMT]&#xA;   X-Ms-Range: [bytes=0-1023]&#xA;   X-Ms-Version: [2019-02-02]&#xA;   --------------------------------------------------------------------------------&#xA;   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.&#xA;   Content-Length: [252]&#xA;   Content-Type: [application/xml]&#xA;   Date: [Mon, 14 Dec 2020 22:08:08 GMT]&#xA;   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]&#xA;   X-Ms-Client-Request-Id: [0dc2b9bc-e974-494d-478c-6da6d0d1c54f]&#xA;   X-Ms-Error-Code: [ConditionNotMet]&#xA;   X-Ms-Request-Id: [1bd3a78a-c01e-0148-5c65-d29926000000]&#xA;   X-Ms-Version: [2019-02-02]&#xA;&#xA;&#xA;</Message><Key>obj1-copy</Key><BucketName>aws-sdk-php-5fd7e24286b7e</BucketName><Resource>/aws-sdk-php-5fd7e24286b7e/obj1-copy</Resource><RequestId>1650B4A53F635C27</RequestId><HostId>c8c20c45-d23a-4ce8-af5e-ff3b0d2296cc</HostId></Error>"
}
(2/15) Running aws-sdk-ruby tests ... FAILED in 10 seconds
{
  "name": "aws-sdk-ruby",
  "function": "getObject(bucket_name, file)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-c87439080e6b",
    "file": "/mint/data/datafile-1-kB",
    "destination": "/tmp"
  },
  "duration": 788.35,
  "error": "-> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:1bd3bb4f-c01e-0148-0965-d29926000000\nTime:2020-12-14T22:08:19.4705522Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/aws-sdk-ruby-bucket-c87439080e6b/datafile-1-kB?timeout=1501\n   Authorization: REDACTED\n   If-Match: [c21b73aa229ee10d81eee0aa2044d4a8]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [eb0c0360-ac70-48f6-5f46-0dc41cc87af4]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:08:19 GMT]\n   X-Ms-Range: [bytes=0-1023]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:08:18 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [eb0c0360-ac70-48f6-5f46-0dc41cc87af4]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [1bd3bb4f-c01e-0148-0965-d29926000000]\n   X-Ms-Version: [2019-02-02]\n\n\n",
  "status": "FAIL"
}
(2/15) Running awscli tests ... FAILED in 8 seconds
{
  "name": "awscli",
  "duration": 3867,
  "function": "aws --endpoint-url http://minio-dev7.minio.io:31994 s3api get-object --bucket awscli-mint-test-bucket-22918 --key datafile-1-kB /tmp/datafile-1-kB\n",
  "status": "FAIL",
  "error": "An error occurred (ConditionNotMet) when calling the GetObject operation: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====Description=The condition specified using HTTP conditional header(s) is not met.RequestId:1bd3ca67-c01e-0148-7865-d29926000000Time:2020-12-14T22:08:26.6529000Z, Details:    Code: ConditionNotMet   GET https://minioazure.blob.core.windows.net/awscli-mint-test-bucket-22918/datafile-1-kB?timeout=1501   Authorization: REDACTED   If-Match: [c21b73aa229ee10d81eee0aa2044d4a8]   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]   X-Ms-Client-Request-Id: [f6803a94-b882-42fa-56cd-eca2b5507298]   X-Ms-Date: [Mon, 14 Dec 2020 22:08:26 GMT]   X-Ms-Range: [bytes=0-1023]   X-Ms-Version: [2019-02-02]   --------------------------------------------------------------------------------   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.   Content-Length: [252]   Content-Type: [application/xml]   Date: [Mon, 14 Dec 2020 22:08:25 GMT]   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]   X-Ms-Client-Request-Id: [f6803a94-b882-42fa-56cd-eca2b5507298]   X-Ms-Error-Code: [ConditionNotMet]   X-Ms-Request-Id: [1bd3ca67-c01e-0148-7865-d29926000000]   X-Ms-Version: [2019-02-02]"
}
(2/15) Running healthcheck tests ... done in 1 seconds
(3/15) Running mc tests ... FAILED in 21 seconds
{
  "name": "mc",
  "duration": "2264",
  "function": "test_get_object",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-140 --quiet --no-color cp myminio/mc-test-bucket-16673/mc-test-object-12050 mc-test-object-12050.downloaded  >>> \n`myminio/mc-test-bucket-16673/mc-test-object-12050` -> `mc-test-object-12050.downloaded`\nmc: <ERROR> Failed to copy `http://minio-dev7.minio.io:31994/mc-test-bucket-16673/mc-test-object-12050`. -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:249e9586-501e-00e3-5b65-d20861000000\nTime:2020-12-14T22:08:49.4756883Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/mc-test-bucket-16673/mc-test-object-12050?timeout=1501\n   Authorization: REDACTED\n   If-Match: [0897fc01942b22f0779aaddcda426858]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [71d71bf6-2ba8-4615-533c-dd5f2219c199]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:08:49 GMT]\n   X-Ms-Range: [bytes=0-1048575]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:08:49 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [71d71bf6-2ba8-4615-533c-dd5f2219c199]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [249e9586-501e-00e3-5b65-d20861000000]\n   X-Ms-Version: [2019-02-02]\n\n\n\nTotal: 0 B, Transferred: 0 B, Speed: 0 B/s"
}
(3/15) Running minio-dotnet tests ... FAILED in 28 seconds
</RecordDelimiter><FieldDelimiter>,</FieldDelimiter></CSV></OutputSerialization></SelectObjectContentRequest>

Unhandled Exception: System.AggregateException: One or more errors occurred. (insufficient data) ---> System.IO.IOException: insufficient data
   at Minio.DataModel.SelectResponseStream.start() in /q/.q/sources/minio-dotnet/Minio/DataModel/Select/SelectResponseStream.cs:line 143
   at Minio.MinioClient.SelectObjectContentAsync(String bucketName, String objectName, SelectObjectOptions opts, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 221
   at Minio.Functional.Tests.FunctionalTest.SelectObjectContent_Test(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2786
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.Program.Main(String[] args) in /mint/run/core/minio-dotnet/Program.cs:line 126
(3/15) Running minio-go tests ... FAILED in 13 seconds
{
  "args": {
    "bucketName": "minio-go-test-qy6mo6nimu1g4nzo",
    "objectName": "wkmcow6553cqt0qvylwndkv9wu1gtk"
  },
  "duration": 974,
  "error": "EOF",
  "function": "GetObject(bucketName, objectName)",
  "message": "Incorrect read bytes v/s original buffer",
  "name": "minio-go: testGetObjectReadSeekFunctionalV2",
  "status": "FAIL"
}
(3/15) Running minio-java tests ... FAILED in 9 minutes and 5 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[multi-part upload]",
  "duration": 2716,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset >>> [java.base/java.net.SocketInputStream.read(SocketInputStream.java:186), java.base/java.net.SocketInputStream.read(SocketInputStream.java:140), okio.InputStreamSource.read(JvmOkio.kt:90), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:129), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.uploadPart(MinioClient.java:4747), io.minio.MinioClient.putObject(MinioClient.java:2738), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testPutObject(FunctionalTest.java:763), FunctionalTest.putObject(FunctionalTest.java:837), FunctionalTest.runObjectTests(FunctionalTest.java:3751), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(3/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 162,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:68:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:684:10) at endReadableNT (_stream_readable.js:1327:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(3/15) Running minio-py tests ... FAILED in 58 seconds
{
  "name": "minio-py:test_get_object",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9",
    "object_name": "be455c69-d96a-4a9a-811f-b5de25a2ccee"
  },
  "message": "S3 operation failed; code: ConditionNotMet, message: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:5b753bb2-101e-0163-3a67-d2ed9e000000\nTime:2020-12-14T22:20:17.8205456Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9/be455c69-d96a-4a9a-811f-b5de25a2ccee?timeout=1501\n   Authorization: REDACTED\n   If-Match: [96f1d3eaf4d06af627a26c38d63df43d]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [570ca2ef-d2a4-454f-6831-bc924a891048]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:20:17 GMT]\n   X-Ms-Range: [bytes=0-1048575]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:20:17 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [570ca2ef-d2a4-454f-6831-bc924a891048]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [5b753bb2-101e-0163-3a67-d2ed9e000000]\n   X-Ms-Version: [2019-02-02]\n\n\n, resource: /minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9/be455c69-d96a-4a9a-811f-b5de25a2ccee, request_id: 1650B54EF00A8AE9, host_id: c8c20c45-d23a-4ce8-af5e-ff3b0d2296cc, bucket_name: minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9, object_name: be455c69-d96a-4a9a-811f-b5de25a2ccee",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 909, in test_get_object\n    _test_get_object(log_entry, sse)\n  File \"/mint/run/core/minio-py/tests.py\", line 896, in _test_get_object\n    bucket_name, object_name, ssec=sse, version_id=version_id,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1101, in get_object\n    preload_content=False,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 408, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 384, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: ConditionNotMet, message: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:5b753bb2-101e-0163-3a67-d2ed9e000000\nTime:2020-12-14T22:20:17.8205456Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9/be455c69-d96a-4a9a-811f-b5de25a2ccee?timeout=1501\n   Authorization: REDACTED\n   If-Match: [96f1d3eaf4d06af627a26c38d63df43d]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [570ca2ef-d2a4-454f-6831-bc924a891048]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:20:17 GMT]\n   X-Ms-Range: [bytes=0-1048575]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:20:17 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [570ca2ef-d2a4-454f-6831-bc924a891048]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [5b753bb2-101e-0163-3a67-d2ed9e000000]\n   X-Ms-Version: [2019-02-02]\n\n\n, resource: /minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9/be455c69-d96a-4a9a-811f-b5de25a2ccee, request_id: 1650B54EF00A8AE9, host_id: c8c20c45-d23a-4ce8-af5e-ff3b0d2296cc, bucket_name: minio-py-test-3a79d932-2ad6-46cd-904b-e989780ea8f9, object_name: be455c69-d96a-4a9a-811f-b5de25a2ccee\n",
  "duration": 2691
}
(3/15) Running s3cmd tests ... FAILED in 1 minutes and 2 seconds
{
  "name": "s3cmd",
  "duration": "1389",
  "function": "test_get_object",
  "status": "FAIL",
  "error": "ERROR: S3 error: 412 (ConditionNotMet): -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=ConditionNotMet) =====\nDescription=The condition specified using HTTP conditional header(s) is not met.\nRequestId:052e9657-401e-003b-2967-d2afb0000000\nTime:2020-12-14T22:21:20.6798849Z, Details: \n   Code: ConditionNotMet\n   GET https://minioazure.blob.core.windows.net/s3cmd-test-bucket-18794/s3cmd-test-object-9883?timeout=1501\n   Authorization: REDACTED\n   If-Match: [0897fc01942b22f0779aaddcda426858]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-14T22:01:48Z]\n   X-Ms-Client-Request-Id: [64b691e1-0569-4b72-6b25-7593c30c9a4d]\n   X-Ms-Date: [Mon, 14 Dec 2020 22:21:20 GMT]\n   X-Ms-Range: [bytes=0-1048575]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 412 The condition specified using HTTP conditional header(s) is not met.\n   Content-Length: [252]\n   Content-Type: [application/xml]\n   Date: [Mon, 14 Dec 2020 22:21:20 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Client-Request-Id: [64b691e1-0569-4b72-6b25-7593c30c9a4d]\n   X-Ms-Error-Code: [ConditionNotMet]\n   X-Ms-Request-Id: [052e9657-401e-003b-2967-d2afb0000000]\n   X-Ms-Version: [2019-02-02]"
}
(3/15) Running s3select tests ... FAILED in 2 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-2703fdc5-74e2-4bde-8a2a-e4ec8c71df51"
  },
  "duration": 1841,
  "message": "Test test_1 unexpectedly failed with: 'event-type'",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/s3select/csv.py\", line 43, in test_sql_api\n    for d in data.stream(10*1024):\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select/reader.py\", line 211, in stream\n    message = self.__extract_message()\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select/reader.py\", line 161, in __extract_message\n    event_type = header_map[\"event-type\"]\nKeyError: 'event-type'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 119, in test_csv_input_custom_quote_char\n    data, sql_opts, expected_output)\n  File \"/mint/run/core/s3select/csv.py\", line 49, in test_sql_api\n    'Test {} unexpectedly failed with: {}'.format(test_name, select_err))\nValueError: Test test_1 unexpectedly failed with: 'event-type'\n",
  "status": "FAIL"
}
(3/15) Running security tests ... done in 1 seconds

Executed 3 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/gateway/azure/gateway-azure.go
+++ b/cmd/gateway/azure/gateway-azure.go
@@ -812,7 +812,7 @@ func (a *azureObjects) GetObjectNInfo(ctx context.Context, bucket, object string
 
 	pr, pw := io.Pipe()
 	go func() {
-		err := a.GetObject(ctx, bucket, object, startOffset, length, pw, objInfo.ETag, opts)
+		err := a.GetObject(ctx, bucket, object, startOffset, length, pw, objInfo.InnerETag, opts)
 		pw.CloseWithError(err)
 	}()
 	// Setup cleanup function to cause the above go-routine to
@@ -833,8 +833,13 @@ func (a *azureObjects) GetObject(ctx context.Context, bucket, object string, sta
 		return azureToObjectError(minio.InvalidRange{}, bucket, object)
 	}
 
+	accessCond := azblob.BlobAccessConditions{}
+	if etag != "" {
+		accessCond.ModifiedAccessConditions.IfMatch = azblob.ETag(etag)
+	}
+
 	blobURL := a.client.NewContainerURL(bucket).NewBlobURL(object)
-	blob, err := blobURL.Download(ctx, startOffset, length, azblob.BlobAccessConditions{}, false)
+	blob, err := blobURL.Download(ctx, startOffset, length, accessCond, false)
 	if err != nil {
 		return azureToObjectError(err, bucket, object)
 	}
@@ -855,6 +860,8 @@ func (a *azureObjects) GetObjectInfo(ctx context.Context, bucket, object string,
 		return objInfo, azureToObjectError(err, bucket, object)
 	}
 
+	realETag := string(blob.ETag())
+
 	// Populate correct ETag's if possible, this code primarily exists
 	// because AWS S3 indicates that
 	//
@@ -866,7 +873,7 @@ func (a *azureObjects) GetObjectInfo(ctx context.Context, bucket, object string,
 	//
 	// Some applications depend on this behavior refer https://github.com/minio/minio/issues/6550
 	// So we handle it here and make this consistent.
-	etag := minio.ToS3ETag(string(blob.ETag()))
+	etag := minio.ToS3ETag(realETag)
 	metadata := blob.NewMetadata()
 	contentMD5 := blob.ContentMD5()
 	switch {
@@ -881,6 +888,7 @@ func (a *azureObjects) GetObjectInfo(ctx context.Context, bucket, object string,
 		Bucket:          bucket,
 		UserDefined:     azurePropertiesToS3Meta(metadata, blob.NewHTTPHeaders(), blob.ContentLength()),
 		ETag:            etag,
+		InnerETag:       realETag,
 		ModTime:         blob.LastModified(),
 		Name:            object,
 		Size:            blob.ContentLength(),

--- a/cmd/gateway/s3/gateway-s3.go
+++ b/cmd/gateway/s3/gateway-s3.go
@@ -431,6 +431,11 @@ func (l *s3Objects) GetObject(ctx context.Context, bucket string, key string, st
 			return minio.ErrorRespToObjectError(err, bucket, key)
 		}
 	}
+
+	if etag != "" {
+		opts.SetMatchETag(etag)
+	}
+
 	object, _, _, err := l.Client.GetObject(ctx, bucket, key, opts)
 	if err != nil {
 		return minio.ErrorRespToObjectError(err, bucket, key)

--- a/cmd/object-api-datatypes.go
+++ b/cmd/object-api-datatypes.go
@@ -168,6 +168,9 @@ type ObjectInfo struct {
 	// Hex encoded unique entity tag of the object.
 	ETag string
 
+	// The ETag stored in the gateway backend
+	InnerETag string
+
 	// Version ID of this object.
 	VersionID string
 

